### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by danfunk**
New dashboard enhancements, including an inventory page

### CODE REVIEW

File: migrations/alembic.ini [hunk 1]
```diff
@@ -43,6 +43,3 @@ formatter = generic
 [formatter_generic]
 format = %(levelname)-5.5s [%(name)s] %(message)s
 datefmt = %H:%M:%S
-
```
Comment (danfunk) [inline migrations/alembic.ini:46]: I added this for a reason.  The scheduler may use the database to do some work, but I don't want it getting processed by alembic.  Why was this removed?

File: communicator/models/sample.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class Sample(db.Model):
     student_id = db.Column(db.Integer)
     computing_id = db.Column(db.String)
     date = db.Column(db.DateTime)
-    last_modified = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.now, nullable=False)
+    last_modified = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.now, nullable=True)
```
Comment (danfunk) [inline communicator/models/sample.py:16]: This field must not be nulable, you may need to downgrade and then upgrade your database, but this should not be altered, I have to guarantee this value is set on all records to allow an external system to pull down information.

File: communicator/services/graph_service.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
+import re
```
Comment (danfunk) [inline communicator/services/graph_service.py:1]: This looks BEAUTIFUL.

File: communicator/templates/pages/imported_files.html [hunk 4]
```diff
@@ -1,25 +1,24 @@
-<!DOCTYPE html>
```
Comment (danfunk) [inline communicator/templates/pages/imported_files.html:1]: Thank you for cleaning this up.

File: migrations/env.py [hunk 5]
```diff
@@ -10,8 +10,6 @@
```
Comment (danfunk) [inline migrations/env.py:10]: Why are you modifying this file?
This exclude stuff was hard to get set up correctly to avoid issues with the scheduler.

File: migrations/versions/087d3afb4e8e_.py [hunk 6]
```diff
@@ -1,31 +0,0 @@
-"""empty message
```
Comment (danfunk) [inline migrations/versions/087d3afb4e8e_.py:1]: Never delete migrations.  I can't roll this out to production.

File: tests/services/test_graph_service.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
+import datetime
```
Comment (danfunk) [inline tests/services/test_graph_service.py:1]: EXCELLENT.

File: migrations/versions/087d3afb4e8e_.py (hunk 6 above)
Comment (NWalker4483) [reply migrations/versions/087d3afb4e8e_.py:1]: I'll copy the migrations over from master and then try to set up my local database with that
